admin/nodes/views.py

```diff
@@ -349,23 +349,34 @@ class NodeFlaggedSpamList(NodeSpamList, DeleteView):
     template_name = 'nodes/flagged_spam_list.html'
 
     def delete(self, request, *args, **kwargs):
-        if not request.user.has_perm('auth.mark_spam'):
+        if (('spam_confirm' in request.POST.keys() and not request.user.has_perm('auth.mark_spam')) or
```

Comment:
Another future proof thing is to wrap any dict iterator in a list like `list(request.POST.keys())` or `list(request.POST.items())`. This is compatible with both Python 2 and 3

Reply:
ah ok, @Johnetordoff, looking at [PEP 3106: Dictionary Views](https://docs.python.org/dev/whatsnew/2.7.html#pep-3106-dictionary-views), will this be required to work with Py3?

Reply:
yes it's already part of the WIP PR, just might as well wrap these in new PRs before we are required to as part of Py3.

Reply:
great, John, thanks for pointing this out!